### PR TITLE
Add test and test cases for create_mapping function

### PR DESCRIPTION
**Pull request by JoerivanEngelen**

### CODE REVIEW

File: tests/test_mapping.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@parametrize_with_cases(
+    "src_idx,tgt_idx,nsrc,ntgt,operator,expected_map_dense,expected_mask"
+)
+def test_create_mapping(
+    src_idx, tgt_idx, nsrc, ntgt, operator, expected_map_dense, expected_mask
+):
+    map_out, mask = create_mapping(src_idx, tgt_idx, nsrc, ntgt, operator)
+
+    assert issubclass(map_out.dtype.type, np.floating)
+    assert issubclass(mask.dtype.type, np.integer)
```
Comment (Hofer-Julian) [inline tests/test_mapping.py:20]: Theoretically we should be able to check that statically with mypy, but maybe that doesn't work yet relieably with numpy.

File: tests/test_mapping.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+from imod_coupler.utils import create_mapping
+
+
+#%% Test
+@parametrize_with_cases(
+    "src_idx,tgt_idx,nsrc,ntgt,operator,expected_map_dense,expected_mask"
+)
+def test_create_mapping(
+    src_idx, tgt_idx, nsrc, ntgt, operator, expected_map_dense, expected_mask
```
Comment (Hofer-Julian) [inline tests/test_mapping.py:15]: Many of the names here are not very obvious

File: tests/test_mapping_cases.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
+import numpy as np
+
+
+def case_1_1_symmetric_sum():
+    src_idx = np.array([0, 1, 2], dtype=int)
+    tgt_idx = np.array([0, 1, 2], dtype=int)
+
+    nsrc: int = 3
+    ntgt: int = 3
```
Comment (Hofer-Julian) [inline tests/test_mapping_cases.py:9]: Is that type hint necessary?

**Review by Hofer-Julian — COMMENTED** [review]
Only small comments, maybe no changes are necessary.

File: tests/test_mapping_cases.py (hunk 3 above)
Comment (JoerivanEngelen) [reply tests/test_mapping_cases.py:9]: Nope, removed!

File: tests/test_mapping.py (hunk 2 above)
Comment (JoerivanEngelen) [reply tests/test_mapping.py:15]: The argument names are equivalent to the `create_mapping` function, I'll write some docstring pointing to the `create_mapping` function.

Comment (Hofer-Julian) [reply tests/test_mapping.py:15]: One more reason to refactor that function :P

**Pull request merged**